### ISSUE TITLE
[Blocker] When fetching file's thumbs through a related fileGroup, all thumbs are the same

### ISSUE THREAD

**Issue by Fen747** [bluelibs/bluelibs#171]
**Describe the bug**
I query a list of comment. Each of them has a file group containing at least one file, and I query the `thumbs` (a custom `small` thumb).

In the database, everything is fine, each file has a thumb with a different `path` and `id` pointing to a different photo.
However, in the query result, though the `files` are all correct with their own `downloadUrl`, they all share the same thumb, which is the older one (first to come in the query result).

**Expected behavior**
Each file should have its own thumbnail.

**Screenshots**
<img width="272" alt="Screen Shot 2022-02-01 at 11 46 31" src="https://user-images.githubusercontent.com/13072245/151954812-36043a59-f742-453e-9daf-98aea0518f6c.png">

**Query result**
```typescript
{
	"CommentsFind": [
		{
			"_id": "61f80edfb5b98a094dc8f371",
			"createdAt": 1643646687362,
			"createdBy": {
				"_id": "615acb7963fde02bc9d13605",
				"fullName": "David Panart",
				"__typename": "User"
			},
			"text": "dsqfqsdf qsdf sqd",
			"attachments": {
				"_id": "61f80edfb5b98a094dc8f370",
				"name": null,
				"files": [
					{
						"_id": "61f80ee4b5b98a094dc8f372",
						"downloadUrl": "https://s3-eu-west-3.amazonaws.com/staging.belett.selfcity/2022/01/31/WvZ60H1lLz-HG8pYBasaZ-Screen Shot 2021-04-02 at 22.04.33.png",
						"mimeType": "image/png",
						"thumbs": [
							{
								"id": "small",
								"path": "2022/01/31/HNOyjFyQul-bP3WgYtVP-Screen Shot 2021-04-02 at 22.04.33.small.png",
								"downloadUrl": "https://s3-eu-west-3.amazonaws.com/staging.belett.selfcity/2022/01/31/HNOyjFyQul-bP3WgYtVP-Screen Shot 2021-04-02 at 22.04.33.small.png",
								"__typename": "AppFileThumb"
							}
						],
						"__typename": "AppFile"
					}
				],
				"__typename": "AppFileGroup"
			},
			"__typename": "Comment"
		},
		{
			"_id": "61f824b121e4f696be9b18e1",
			"createdAt": 1643652273302,
			"createdBy": {
				"_id": "61e1867613c0e556923dae1a",
				"fullName": "Jean-Baptiste FLEURY",
				"__typename": "User"
			},
			"text": "testaze aze aze aze",
			"attachments": {
				"_id": "61f824b121e4f696be9b18e0",
				"name": null,
				"files": [
					{
						"_id": "61f824b321e4f696be9b18e2",
						"downloadUrl": "https://s3-eu-west-3.amazonaws.com/staging.belett.selfcity/2022/01/31/GABMKGrW26-aRm_M94C6f-16427900095784808577899809384774.jpg",
						"mimeType": "image/jpeg",
						"thumbs": [
							{
								"id": "small",
								"path": "2022/01/31/mT2OLrh5eP-D-DVNXMY7-16427900095784808577899809384774.small.jpg",
								"downloadUrl": "https://s3-eu-west-3.amazonaws.com/staging.belett.selfcity/2022/01/31/mT2OLrh5eP-D-DVNXMY7-16427900095784808577899809384774.small.jpg",
								"__typename": "AppFileThumb"
							}
						],
						"__typename": "AppFile"
					}
				],
				"__typename": "AppFileGroup"
			},
			"__typename": "Comment"
		},
		{
			"_id": "61f824ea21e4f696be9b18e4",
			"createdAt": 1643652330974,
			"createdBy": {
				"_id": "61e1867613c0e556923dae1a",
				"fullName": "Jean-Baptiste FLEURY",
				"__typename": "User"
			},
			"text": "aze aze aze aze aeD QSFSD F",
			"attachments": {
				"_id": "61f824ea21e4f696be9b18e3",
				"name": null,
				"files": [
					{
						"_id": "61f824eb21e4f696be9b18e5",
						"downloadUrl": "https://s3-eu-west-3.amazonaws.com/staging.belett.selfcity/2022/01/31/nHcBp1cfZH-RII_Km3n9b-LOGO PICTO RW.png",
						"mimeType": "image/png",
						"thumbs": [
							{
								"id": "small",
								"path": "2022/01/31/LlfTSp4p2M-exdc1a3a--LOGO PICTO RW.small.png",
								"downloadUrl": "https://s3-eu-west-3.amazonaws.com/staging.belett.selfcity/2022/01/31/LlfTSp4p2M-exdc1a3a--LOGO PICTO RW.small.png",
								"__typename": "AppFileThumb"
							}
						],
						"__typename": "AppFile"
					}
				],
				"__typename": "AppFileGroup"
			},
			"__typename": "Comment"
		}
	]
}
```

**Comment by Fen747** [comment]
Ok, so it seems the query result on the network is OK, `BUT` the list of comments I get from the `useData` has all the `thumbs` be the same :

```typescript
    const {
      data: comments,
      isLoading,
      error,
    } = useData(CommentsCollection, queryInput, {
      _id: 1,
      createdAt: 1,
      createdBy: { _id: 1, fullName: 1 },
      text: 1,
      attachments: {
        _id: 1,
        name: 1,
        files: {
          _id: 1,
          downloadUrl: 1,
          mimeType: 1,
          thumbs: {
            id: 1,
            path: 1,
            downloadUrl: 1,
          },
        },
      },
    });

    console.log(
      "comments______",
      comments && comments.map((c) => c.attachments),
    );
```


The log :

<img width="1178" alt="Screen Shot 2022-02-01 at 17 37 44" src="https://user-images.githubusercontent.com/13072245/152010637-3a24b299-ef0d-48e0-aa6c-b95bb1f66414.png">


I'm looking if it might comes from client-side collection's transform or something like this

**Comment by theodorDiaconu** [comment]
Now that is odd, did you do anything specific to your client-side colls?

**Comment by Fen747** [comment]
Nothing.

**Comment by theodorDiaconu** [comment]
Released x-s3-bundle@1.1.0

The problem was from having 'id' which interfered with ApolloCache showing the same result. We've made a `type` field available in the thumbnail and deprecated id with a warning for GraphQL only. Nothing changes on the backend. You won't be interacting with `id` that much so we'll leave them untouched for now to not include any BC breaks or complex upgrade paths.
